experiment/simulation/js/fa.js

```diff
@@ -0,0 +1,333 @@
+import { registerGate } from "./main.js";
+import { setPosition } from "./layout.js";
+import {gates} from './gate.js';
+import {jsPlumbInstance} from "./main.js";
+
+// Dictionary of all full adders in the circuit with their IDs as keys
+let fullAdder = {}
+
+function clearFAs() {
+
+    for (let faID in fullAdder) {
+        delete fullAdder[faID];
+    }
+    fullAdder = {};
+}
+
+// {output-id: [gate,pos]}
+let finalOutputs = {
+    "Output-5": [],
+    "Output-8": [],
+    "Output-11": [],
+    "Output-12": [],
+};
+
+class FullAdder {
+    constructor() {
+        this.id = "FullAdder-" + window.numComponents++;
+        this.a0 = [];  // Takes 2 items in a list : Gate, Output endpoint of gate
+        this.b0 = [];
+        this.Cin = [];
+        this.sum = null;
+        this.Cout = null;
+        this.inputPoints = [];
+        this.outputPoints = [];
+        this.CoutIsConnected = false;
+        this.sumIsConnected = false;
+        this.component = '<div class="drag-drop FullAdder" id=' + this.id + ' style="width:100px;height:100px;"></div>';
+    }
+
+    // Adds element to the circuit board, adds event listeners and generates its endpoints.
+    registerComponent(workingArea,x = 0, y = 0) {
+        const parent = document.getElementById(workingArea);
+        parent.insertAdjacentHTML('beforeend', this.component);
+        document.getElementById(this.id).style.left = x + "px";
+        document.getElementById(this.id).style.top = y + "px";
+
+        const el = document.getElementById(this.id);
+        el.addEventListener('contextmenu', function (ev) {
+            ev.preventDefault();
+                let left = ev.pageX - document.getScroll()[0];
+                let top = ev.pageY - document.getScroll()[1];

```

Comment:
Please either make left and top const or just remove them and do the calculation right inside the definition of origin.

Reply:
Okay, done